app/src/components/search/MetabaseSearch.tsx

```diff
@@ -0,0 +1,137 @@
+import DateFnsUtils from '@date-io/date-fns';
+import { Button, Grid, List, ListItem, makeStyles, Paper, TextField } from '@material-ui/core';
+import { Add, DeleteForever } from '@material-ui/icons';
+import { MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { DatabaseChangesContext } from 'contexts/DatabaseChangesContext';
+import { DatabaseContext } from 'contexts/DatabaseContext';
+import React, { useContext, useEffect, useState, useCallback } from 'react';
+
+interface IMetabaseChoices {
+  metabaseQueryId: string;
+}
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    padding: theme.spacing(2),
+    textAlign: 'center',
+    color: theme.palette.text.secondary
+  },
+  metabaseRecordsChoice: {
+    padding: theme.spacing(2)
+  },
+  metabaseSearchField: {
+    width: '250px'
+  },
+  metabaseFilter: {
+    flexDirection: 'column'
+  },
+  metabaseAddButton: {
+    width: '100%'
+  }
+}));
+
+export const MetabaseSearch: React.FC<any> = (props) => {
+  const databaseContext = useContext(DatabaseContext);
+  const databaseChangesContext = useContext(DatabaseChangesContext);
+  const [metabaseChoices, setMetabaseChoices] = useState([]);
+
+  const getMetabaseChoicesFromTrip = useCallback(async () => {
+    let docs = await databaseContext.database.find({
+      selector: {
+        _id: 'trip'
+      }
+    });
+    if (docs.docs.length > 0) {
+      let tripDoc = docs.docs[0];
+      if (tripDoc.metabaseChoices) {
+        setMetabaseChoices([...tripDoc.metabaseChoices]);
+      }
+    }
+  }, [databaseContext.database]);
+
+  useEffect(() => {
+    const updateComponent = () => {
+      getMetabaseChoicesFromTrip();
+    };
+    updateComponent();
+  }, [databaseChangesContext, getMetabaseChoicesFromTrip]);
+
+  const saveChoices = async (newMetabaseChoices) => {
+    await databaseContext.database.upsert('trip', (tripDoc) => {
+      return { ...tripDoc, metabaseChoices: newMetabaseChoices };
+    });
+  };
+
+  const addMetabaseChoice = (newMetabase: IMetabaseChoices) => {
+    saveChoices([...metabaseChoices, newMetabase]);
+  };
+
+  const updateMetabaseChoice = (updatedMetabase: IMetabaseChoices, index: number) => {
+    let updatedMetabaseChoices = [...metabaseChoices];
```

Comment:
can this be a const?
```suggestion
    const updatedMetabaseChoices = [...metabaseChoices];
```